src/main/java/com/weirddev/testme/intellij/template/context/Field.java

```diff
@@ -42,21 +54,55 @@ public class Field {
     public Field(PsiField psiField, PsiClass srcClass, TypeDictionary typeDictionary, int maxRecursionDepth) {
         this.name = psiField.getName();
         type= buildType(psiField.getType(), typeDictionary, maxRecursionDepth);
+        this.isAnnotatedByDI = buildAnnotatedByDI(psiField);
+        this.hasSetter = buildHasSetter(srcClass.getMethods(), psiField.getName());
         String canonicalText = srcClass.getQualifiedName();
         ownerClassCanonicalName = ClassNameUtils.stripArrayVarargsDesignator(canonicalText);
         overridden = isOverriddenInChild(psiField, srcClass);
         isFinal = psiField.getModifierList() != null && psiField.getModifierList().hasExplicitModifier(PsiModifier.FINAL);
         isStatic = psiField.getModifierList() != null && psiField.getModifierList().hasExplicitModifier(PsiModifier.STATIC);
     }
 
-    private Type buildType(PsiType type, TypeDictionary typeDictionary, int maxRecursionDepth) {
+    /**
+     * @param type PsiType
+     * @param typeDictionary type dictionary
+     * @param maxRecursionDepth recursion depth
+     * @return the Type from PsiType
+     */
+    private static Type buildType(PsiType type, TypeDictionary typeDictionary, int maxRecursionDepth) {
         if (typeDictionary == null) {
             return new Type(type, null, null, 0, false);
         } else {
             return typeDictionary.getType(type, maxRecursionDepth, true);
         }
     }
 
+    /**
+     * 
+     * @param methods psi methods of tested class
+     * @param fieldName field of tested class
+     * @return true if field has setter
+     */
+    private boolean buildHasSetter(PsiMethod[] methods, String fieldName) {
+        return null != methods && methods.length > 0
+            && Arrays.stream(methods).anyMatch(psiMethod -> PropertyUtils.isPropertySetter(psiMethod, fieldName));
+    }
+
+    /**
+     *
+     * @return true if the field annotations (Like @Resource, @Autowired) indicates it is di field
+     */
+    private boolean buildAnnotatedByDI(PsiField psiField) {
+        PsiAnnotation[] fieldAnnotations = psiField.getAnnotations();
+        return null != fieldAnnotations && fieldAnnotations.length > 0 && Arrays.stream(fieldAnnotations)
+            .anyMatch(this::isDiFieldAnnotation);
```

Comment:
on each call - isDiFieldAnnotation() creates a collection and loops it. to improve performance, create a class with static member - `Set<String> DiFieldAnnotationTypes` - initialized once to the enum values property _cannonicalName_  - use it to check if `Set#contains(psiAnnotation.getQualifiedName())`.
likewise, same class can hold also a static Set for the matching Class level DI annotations type names for a similar use case in _Type_